### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by edmundnoble**
Depends on #5. Run with `it/run`. Not yet documented.

### CODE REVIEW

File: it/src/main/scala/quasar/physical/s3/LWCTests.scala [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+import pathy._
+import quasar.Data
+import quasar.contrib.pathy._
+import quasar.fs.mount.ConnectionUri
+import quasar.mimir.LightweightFileSystem
+import scalaz._, Scalaz._
+import scalaz.concurrent.Task
+
+final case class TestFS(
```
Comment (alissapajer) [inline it/src/main/scala/quasar/physical/s3/LWCTests.scala]: Is there a reason you have this file in `it/src/main/` rather than `it/src/test/`?

Comment (alissapajer) [reply it/src/main/scala/quasar/physical/s3/LWCTests.scala]: Also curious why you designed this with a `main` method, and not using `specs2`?

**Comment by alissapajer** [comment]
It would be useful to separate the components of this test that are S3-specific from the components that  are not. Then we can provide a property checks suite to anyone writing a connector. And for that reason, I think avoiding the `specs2` dep makes sense :) 

Though make sense to factor this out like that after we've written a couple more integration tests for specific lwc's. Do we want the general test infrastructure (not connector dependent) in its own repo or in quasar?

File: it/src/main/scala/quasar/physical/s3/LWCTests.scala (hunk 1 above)
Comment (alissapajer) [reply it/src/main/scala/quasar/physical/s3/LWCTests.scala]: I've basically answered these two questions for myself after thinking about it a little more :)

**Comment by edmundnoble** [comment]
@alissapajer I don't see any need for property testing here. I *initially* looked at that approach, but realized:
a) The only law-like thing I have is "paths returned by `children` must also return true for `exists`"; but using randomly generated paths makes no sense here
b) The tests don't depend on generated data; all we do is pull down the full filesystem, compare it with what we believe should be present, and run some extra `exists` calls.

Also, what we have in here is entirely non-specific to S3; all that's connector-specific is the definition of "expected" `TestFS` values.

I believe that as long as we aren't factoring out other, more change-sensitive things (like `assembleLWC`) it makes sense to do what I have in the LWC template right now: https://github.com/edmundnoble/lwc.g8/blob/master/src/main/g8/it/src/main/scala/quasar/physical/LWCTests.scala

**Comment by alissapajer** [comment]
I'm :+1: to not factoring this out yet; that's just something that at some point I think will make sense, so we can continue to consider when to do it as work progresses.

**Pull request merged**